### PR TITLE
task/take_picture/use_camera

### PR DESCRIPTION
**Pull request by teamagam5**
Added functionality of taking pictures

### CODE REVIEW

File: DailySelfie/app/src/main/java/com/teamagam/dailyselfie/MainActivity.java [hunk 1]
```diff
@@ -1,12 +1,29 @@
...
+
+import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
 
 public class MainActivity extends AppCompatActivity {
 
+    static final int REQUEST_TAKE_PHOTO = 1;
```
Comment (teamagam1) [inline DailySelfie/app/src/main/java/com/teamagam/dailyselfie/MainActivity.java]: private

**Review by teamagam1 — CHANGES_REQUESTED** [review]

File: DailySelfie/app/src/main/java/com/teamagam/dailyselfie/MainActivity.java [hunk 2]
```diff
@@ -19,4 +36,67 @@ public boolean onCreateOptionsMenu(Menu menu) {
...
+    private void dispatchTakePictureIntent() {
+        Intent takePictureIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+        // Ensure that there's a camera activity to handle the intent
+        if (takePictureIntent.resolveActivity(getPackageManager()) != null) {
+            // Create the File where the photo should go
+            File photoFile = null;
+            try {
+                photoFile = createImageFile();
+            } catch (IOException e) {
+                Toast.makeText(this, e.getMessage(), Toast.LENGTH_SHORT).show();
```
Comment (teamagam1) [inline DailySelfie/app/src/main/java/com/teamagam/dailyselfie/MainActivity.java]: are you sure you want all the stack trace in the Toast message?

File: DailySelfie/app/src/main/java/com/teamagam/dailyselfie/MainActivity.java [hunk 3]
```diff
@@ -19,4 +36,67 @@ public boolean onCreateOptionsMenu(Menu menu) {
...
+            }
+            // Continue only if the File was successfully created
+            if (photoFile != null) {
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
```
Comment (teamagam1) [inline DailySelfie/app/src/main/java/com/teamagam/dailyselfie/MainActivity.java:63]: why do address this SDK?
have you checked the app in this SDK?

you need to be focused on min SDK and max SDK (.N is way over the scope of this project),

File: DailySelfie/app/src/main/java/com/teamagam/dailyselfie/MainActivity.java [hunk 4]
```diff
@@ -19,4 +36,67 @@ public boolean onCreateOptionsMenu(Menu menu) {
...
+            }
+        }
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+
+        if (requestCode == REQUEST_TAKE_PHOTO && resultCode == RESULT_OK) {
+            Toast.makeText(this, getString(R.string.activity_main_picture_taken) + "\n" + mCurrentPhotoPath, Toast.LENGTH_LONG).show();
```
Comment (teamagam1) [inline DailySelfie/app/src/main/java/com/teamagam/dailyselfie/MainActivity.java]: you're using mCurrentPhotoPath field.

but how do you sure this is not null? (think about it)

Comment (teamagam1) [inline DailySelfie/app/src/main/java/com/teamagam/dailyselfie/MainActivity.java]: also, you dont need to keep in Git your private test code. (this exact toast message with full path of the file is not what the user is looking for)

**Review by teamagam1 — CHANGES_REQUESTED** [review]

File: DailySelfie/app/src/main/java/com/teamagam/dailyselfie/MainActivity.java (hunk 2 above)
Comment (teamagam5) [reply DailySelfie/app/src/main/java/com/teamagam/dailyselfie/MainActivity.java]: Didn't want to handle errors at all but had to, so did it poorly.
I'll fix that with simply toasting "IOexception occured" OK?

File: DailySelfie/app/src/main/java/com/teamagam/dailyselfie/MainActivity.java (hunk 3 above)
Comment (teamagam5) [reply DailySelfie/app/src/main/java/com/teamagam/dailyselfie/MainActivity.java:63]: I also have a Nexus 6P emulator so I use it to check some specific features (the camera for example works better on this emulator)

File: DailySelfie/app/src/main/java/com/teamagam/dailyselfie/MainActivity.java (hunk 4 above)
Comment (teamagam5) [reply DailySelfie/app/src/main/java/com/teamagam/dailyselfie/MainActivity.java]: OK. Thought you will use it...

File: DailySelfie/app/src/main/java/com/teamagam/dailyselfie/MainActivity.java (hunk 2 above)
Comment (teamagam5) [reply DailySelfie/app/src/main/java/com/teamagam/dailyselfie/MainActivity.java]: Fix: I'll use toString()

File: DailySelfie/app/src/main/java/com/teamagam/dailyselfie/MainActivity.java (hunk 4 above)
Comment (teamagam5) [reply DailySelfie/app/src/main/java/com/teamagam/dailyselfie/MainActivity.java]: Anyway I removed it

File: DailySelfie/app/src/main/java/com/teamagam/dailyselfie/MainActivity.java (hunk 2 above)
Comment (teamagam5) [reply DailySelfie/app/src/main/java/com/teamagam/dailyselfie/MainActivity.java]: Fix2: informative toast will appear

**Pull request merged**